Makefile

```diff
@@ -114,6 +116,22 @@ PLUGIN_PROTOS= \
 	google/protobuf/descriptor.proto \
 	swift-options.proto
 
+XCODE_BUILD_EXTRAS =
+# Invoke make with XCODE_SKIP_OPTIMIZER=1 to suppress the optimizer when
+# building the Xcode projects. For Release builds, this is a non trivial speed
+# up for compilation
+XCODE_SKIP_OPTIMIZER=0
+ifeq "$(XCODE_SKIP_OPTIMIZER)" "1"
+  XCODE_BUILD_EXTRAS += SWIFT_OPTIMIZATION_LEVEL=-Onone
+endif
+
+# Invoke make with XCODE_ANALYZE=1 to enable the analyzer while building the
+# Xcode projects.
+XCODE_ANALYZE=0
+ifeq "$(XCODE_ANALYZE)" "1"
+  XCODE_BUILD_EXTRAS += RUN_CLANG_STATIC_ANALYZER=YES CLANG_STATIC_ANALYZER_MODE=deep
+endif
```

Comment:
Same, we could change things so the default it to do this and one has to skip it instead.